### DISCUSSION TITLE
pmap changes the behaviour of a NumPyro sampler

### DISCUSSION THREAD

**Discussion by jeremiecoullon** [jax-ml/jax#11109] [Q&A]
Hello!

I documented [this behaviour](https://github.com/pyro-ppl/numpyro/issues/1427) in the NumPyro issue tracker. Essentially I found that in some cases, `pmap(f)(x[None])[0]` and `f(x)` do not produce the same result.

I was wondering if this behaviour is expected from `pmap`? And is it perhaps due to numerical precision errors?

**Comment by sharadmv** [comment]
It can vary from backend to backend (CPU vs GPU vs TPU) but in general, JIT compilation (including pmap) can change the numerics of a function due to compiler optimizations.

What backend are you using?

**Reply by jeremiecoullon** [reply]
Thanks for your answer, and sorry I never answered this!

I'm using CPU as backend. So in this context, changing the numerics of a function due to compiler optimizations can lead to numerical errors. Is that what your mean?
